### PR TITLE
Первый пробный Pull request

### PR DESCRIPTION
**Pull request by MaximSemenov**
Это первый пробный Pull request, в нем есть некоторые незначительные изменения оригинального кода.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,8 +24,12 @@
     "@angular/platform-browser": "^5.2.0",
     "@angular/platform-browser-dynamic": "^5.2.0",
     "@angular/router": "^5.2.0",
+    "angular-bootstrap-md": "^6.0.1",
```
Comment (stevermeister) [inline package.json]: remove

File: src/app/car-tax-form/car-tax-form.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (stevermeister) [inline src/app/car-tax-form/car-tax-form.component.spec.ts]: do not commit empty tests, please

File: src/app/car-tax-form/car-tax-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      fuelType: new FormControl('', Validators.required),
+      volume: new FormControl('', Validators.required)
+    });
+
+    // this.carTaxControl.statusChanges.filter(Boolean).subscribe(console.log);
+    this.carTaxControl.valueChanges.subscribe(console.log);
+
+    this.carTaxControl.statusChanges.filter((status: string) => status === 'VALID').subscribe(() => {
+
+      const value = this.carTaxControl.value;
```
Comment (stevermeister) [inline src/app/car-tax-form/car-tax-form.component.ts]: you can get value from subscription `valueChanges`

File: src/app/car-tax-form/car-tax.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { TestBed, inject } from '@angular/core/testing';
```
Comment (stevermeister) [inline src/app/car-tax-form/car-tax.service.spec.ts]: also remove

File: src/app/car-tax-form/car-tax.service.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { HttpClient } from '@angular/common/http';
+
+
+export type FuelTypes = string[];
+export type Provinces = {
+  'key': string;
+  'title': string;
+};
+
+export type Grid = {
```
Comment (stevermeister) [inline src/app/car-tax-form/car-tax.service.ts]: you can change data structure if you want

File: src/app/car-tax-form/car-tax.service.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+@Injectable()
+export class CarTaxService {
+
+  constructor(private _http: HttpClient) {
+  }
+
+  getFuelTypes(): Observable<FuelTypes> {
+
+    return this._http.get<FuelTypes>('./assets/json/fuelTypes.json');
```
Comment (stevermeister) [inline src/app/car-tax-form/car-tax.service.ts]: not sure whether you need to load static data that is already known, you can compile it into your app

File: tsconfig.json [hunk 7]
```diff
@@ -1,4 +1,8 @@
 {
+  "include": [
+    "node_modules/angular-bootstrap-md/**/*.ts",
```
Comment (stevermeister) [inline tsconfig.json]: remove

File: .gitignore [hunk 8]
```diff
@@ -1,5 +1,11 @@
 # See http://help.github.com/ignore-files/ for more about ignoring files.
 
+# empty tests
```
Comment (stevermeister) [inline .gitignore]: why do you need them in gitignore?

Comment (MaximSemenov) [reply .gitignore]: I am was trying to get rid of them.

**Comment by MaximSemenov** [comment]
screen shot 2018-04-27 at 12 17 36 am

**Comment by stevermeister** [comment]
👍

**Pull request merged**